client-spark/common/src/main/java/org/apache/spark/shuffle/writer/BufferManagerOptions.java

```diff
@@ -64,10 +64,13 @@ public BufferManagerOptions(SparkConf sparkConf) {
             RssSparkConfig.RSS_WRITER_PRE_ALLOCATED_BUFFER_SIZE.defaultValue().get());
     requireMemoryInterval = sparkConf.get(RssSparkConfig.RSS_WRITER_REQUIRE_MEMORY_INTERVAL);
     requireMemoryRetryMax = sparkConf.get(RssSparkConfig.RSS_WRITER_REQUIRE_MEMORY_RETRY_MAX);
-    LOG.debug(RssSparkConfig.RSS_WRITER_BUFFER_SIZE.key() + "=" + bufferSize);
-    LOG.debug(RssSparkConfig.RSS_WRITER_BUFFER_SPILL_SIZE.key() + "=" + bufferSpillThreshold);
-    LOG.debug(
-        RssSparkConfig.RSS_WRITER_PRE_ALLOCATED_BUFFER_SIZE.key() + "=" + preAllocatedBufferSize);
+    if (LOG.isDebugEnabled()) {
```

Comment:
There is no benefit of checking `LOG.isDebugEnabled())` in conjunction with string interpolation. Calling `LOG.debug` when debug level is not enabled will be a NOOP (takes no time). Problem is using an argument like `RssSparkConfig.RSS_WRITER_BUFFER_SIZE.key() + "=" + bufferSize`, which will be computed **before** `LOG.debug` is being called.

Whereas string interpolation like `LOG.debug("{}={}", RssSparkConfig.RSS_WRITER_BUFFER_SIZE.key(), bufferSize)` will not compute the message before calling into `LOG.debug`.